cmd/up/up.go

```diff
@@ -352,6 +352,12 @@ func (up *upContext) activate(autoDeploy, build, resetSyncthing bool) {
 
 		if prevError != nil {
 			if up.shouldRetry(prevError) {
```

Comment:
this has a lot of nested conditions, it makes it hard to understand what's happening. maybe moving it a separate function can allow us to simplify it? (like a `retry` function that checks if the pod needs to be destroyed, sets the retry flag, and other things we do in previous lines of this function)

Reply:
and maybe add the preverErr != nil condition to `shouldRetry` as well